confluent_kafka/src/confluent_kafka.c

```diff
@@ -249,6 +249,39 @@ static PyObject* KafkaError_richcompare (KafkaError *self, PyObject *o2,
 	return result;
 }
 
+static PyObject *KafkaError_new (PyTypeObject *type, PyObject *args,
+                                 PyObject *kwargs) {
+        return type->tp_alloc(type, 0);
+}
+
+/**
+ * @brief Initialize a KafkaError object.
+ */
+static void KafkaError_init (KafkaError *self,
+			     rd_kafka_resp_err_t code, const char *str) {
+	self->code = code;
+        self->fatal = 0;
+        self->retriable = 0;
+        self->txn_requires_abort = 0;
+	if (str)
+		self->str = strdup(str);
+	else
+		self->str = NULL;
+}
+
+static int KafkaError_init0 (PyObject *selfobj, PyObject *args, PyObject *kwargs) {
+        KafkaError *self = (KafkaError *)selfobj;
+        int code;
+        const char *reason = NULL;
+ 
+        static char *kws[] = { "error", "reason",  NULL };
```

Comment:
It would be good to add optional retriable, fatal, txn_requires_abort so that those errors can be mocked.
This also allows us to remove the test_create_fatal_error() or whatever it is called.

Reply:
rename error to error_code or code.

Reply:
in that order?

Reply:
Doesn't really matter, but let's do fatal, retriable, txn_requires_abort to match what is in init().

Reply:
:+1: